libs/ledgerjs/packages/hw-app-btc/src/Btc.ts

```diff
@@ -263,25 +271,31 @@ export default class Btc {
     // if BtcOld was instantiated, stick with it
     if (this._impl instanceof BtcOld) return this._impl;
 
-    const appAndVersion = await getAppAndVersion(this._transport);
-    let isBtcLegacy = true; // default for all altcoins
+    const { name, version } = await getAppAndVersion(this._transport);
 
-    if (appAndVersion.name === "Bitcoin" || appAndVersion.name === "Bitcoin Test") {
-      const [major, minor] = appAndVersion.version.split(".");
-      // we use the legacy protocol for versions below 2.1.0 of the Bitcoin app.
-      isBtcLegacy = parseInt(major) <= 1 || (parseInt(major) == 2 && parseInt(minor) == 0);
-    } else if (
-      appAndVersion.name === "Bitcoin Legacy" ||
-      appAndVersion.name === "Bitcoin Test Legacy"
-    ) {
-      // the "Bitcoin Legacy" and "Bitcoin Testnet Legacy" app use the legacy protocol, regardless of the version
-      isBtcLegacy = true;
-    } else if (appAndVersion.name === "Exchange") {
-      // We can't query the version of the Bitcoin app if we're coming from Exchange;
-      // therefore, we use a workaround to distinguish legacy and new versions.
-      // This can be removed once Ledger Live enforces minimum bitcoin version >= 2.1.0.
-      isBtcLegacy = await checkIsBtcLegacy(this._transport);
-    }
+    const isBtcLegacy = await (async () => {
+      switch (name) {
+        case "Bitcoin":
+        case "Bitcoin Test": {
+          // we use the legacy protocol for versions below 2.1.0 of the Bitcoin app.
+          return semver.lt(version, "2.1.0");
```

Comment:
Are you 100% sure this does the same as the current implem?

Just being paranoid, because app versions can be different from a clean `major.minor.patch`, e.g they can have `-rc0` `-next0` or such additional stuff at the end, and it happened in the past that semver doesn't behave as we think it does.

Reply:
I think the code that he writes is good. There was a bug of the semver lib long time ago and it has been fixed. @haammar-ledger if you have a concrete scenario, we can test it

Reply:
Seems fixed indeed. @haammar-ledger I added these tests just in case 3849cfc44119f04945f0aee5d51e6b25b35c59dd

They are testing the semver lib implementation so should not be done in our codebase, but can be a way of protecting ourselves against unexpected behaviour from semver on these cases (cf. comment)

Feel free to remove this commit if you feel it has nothing to do in our codebase